pytest_plugins/jira_comments.py

```diff
@@ -81,21 +82,58 @@ def pytest_runtest_makereport(item, call):
             update_issue_to_tests_map(item, blocked_by_issues, test_result)
 
 
+def escape_special_characters(text):
+    special_chars = '[]'
+    for char in special_chars:
+        text = text.replace(char, f'\\{char}')
+    return text
+
+
 def pytest_sessionfinish(session, exitstatus):
     """Add test result comment to related Jira issues."""
     if hasattr(session.config, 'issue_to_tests_map'):
         user = os.environ.get('USER')
         build_url = os.environ.get('BUILD_URL')
         for issue in session.config.issue_to_tests_map:
+            # Sort test result based on the outcome.
+            session.config.issue_to_tests_map[issue].sort(key=lambda x: x['outcome'])
+            all_tests_passed = True
             comment_body = (
                 f'This is an automated comment from job/user: {build_url if build_url else user} for a Robottelo test run.\n'
                 f'Satellite/Capsule: {settings.server.version.release} Snap: {settings.server.version.snap} \n'
                 f'Result for tests linked with issue: {issue} \n'
             )
             for item in session.config.issue_to_tests_map[issue]:
-                comment_body += f'{item["nodeid"]} : {item["outcome"]} \n'
+                color_code = '{color:green}'
+                if item["outcome"] == 'failed':
+                    all_tests_passed = False
+                    color_code = '{color:red}'
+                # Color code test outcome
+                color_coded_result = f'{color_code}{item["outcome"]}{{color}}'
+                # Escape special characters in the node_id.
+                escaped_node_id = escape_special_characters(item["nodeid"])
+                comment_body += f'{escaped_node_id} : {color_coded_result} \n'
             try:
-                add_comment_on_jira(issue, comment_body)
+                labels = (
+                    [{'add': JIRA_TESTS_PASSED_LABEL}, {'remove': JIRA_TESTS_FAILED_LABEL}]
```

Comment:
Is it requested from JIRA APIs to feed in this format where its a list of dictionaries, could this accept in the following format?:

```suggestion
                    {'add': JIRA_TESTS_PASSED_LABEL, 'remove': JIRA_TESTS_FAILED_LABEL}
```
if yes, please do, else just resolve the comment without any action.

Non-blocking comment.

Reply:
yes, it's based on JIRA APIs:/